cimgraph/data_profile/cimhub_2023/cimhub_2023.py

```diff
@@ -0,0 +1,18692 @@
+from __future__ import annotations
+
+import json
+import logging
+from dataclasses import dataclass, field, is_dataclass
+from enum import Enum
+from random import Random
+from typing import Optional
+from uuid import UUID, uuid4
+
+_log = logging.getLogger(__name__)
+'''
+    Annotated CIMantic Graphs data profile for cim17v40
+    Generated by CIMTool http://cimtool.org
+'''
+@dataclass
+class Identity():
+    '''
+    This is the new root class from CIM 18 to provide common identification
+    for all classes needing identification and naming attributes.
+    IdentifiedObject is now a child class of Identity.
+    mRID is superseded by Identity.identifier, which is typed to be a UUID.
+    '''
+    identifier: Optional[UUID] = field(
+        default = None,
+        metadata = {
+            'type': 'Attribute',
+            'minOccurs': '1',
+            'maxOccurs': '1'
+        })
+
+    # Backwards support for objects created with mRID
+    def __post_init__(self) -> None:
```

Comment:
As I see it there are a couple reasons to call self.uuid() in post_init.

1. self.identifier is None.
2. self.identifer doesn't match self.mRID.

Currently if the dataclass instance has self.mRID it calls self.uuid() no matter what. This call is completely unnecessary if a user specified identifier and mRID such that they match and the is the source of the bug I'm seeing where identifier if different from what I specified in the constructor.